### PR TITLE
Remove erroneously added `wildcard` method name

### PR DESCRIPTION
**Pull request by heysweet**
resolves https://github.com/dbt-labs/dbt-core/issues/9641

### Problem

https://github.com/dbt-labs/dbt-core/pull/6599/files#diff-9aa4de7075574ef36d838665148e5858294fae6fd66c0eed1fcf2033a34ee84eR50 erroneously added a new key to the `MethodName` enum which is completely unused -- all wildcard operations are performed _within_ other methods, specifically:
* fqn
* tag
* source
* exposure
* metric
* file
* package
* test

### Solution

Remove the unused key to avoid polluting the `MethodName` enum.

### Before

The behavior before this PR  is **not** what we want:

```shell
$ dbt test -s 'wildcard:something'

12:49:35  Running with dbt=1.8.0-b1
12:49:35  Registered adapter: postgres=1.8.0-b1
12:49:36  Found 1 model, 411 macros
12:49:36  Encountered an error:
Internal Error
  Method name "wildcard" is a valid node selection method name, but it is not handled
```

### After

The behavior after this PR **is** what we want:

```shell
$ dbt test -s 'wildcard:something'

12:50:29  Running with dbt=1.8.0-b1
12:50:29  Registered adapter: postgres=1.8.0-b1
12:50:30  Found 1 model, 411 macros
12:50:30  Encountered an error:
Runtime Error
  'wildcard' is not a valid method name
```

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [X] I have run this code in development and it appears to resolve the stated issue  
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [X] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

**Comment by heysweet** [comment]
> Thanks for your pull request, and welcome to our community! We require contributors to sign our [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScfOV7K4enYRHozrDRP6BBIXjOij-JDGca6WBTHyP_ANXSqlg/viewform) and we don't seem to have your signature on file. Check out [this article](https://docs.getdbt.com/docs/contributor-license-agreements) for more information on why we have a CLA.
> 
> In order for us to review and merge your code, please submit the Individual Contributor License Agreement form attached above above. If you have questions about the CLA, or if you believe you've received this message in error, please reach out through a comment on this PR.
> 
> CLA has not been signed by users: @heysweet

I just signed the CLA

**Review by dbeatty10 — APPROVED** [review]
Thanks for noticing and fixing this @heysweet !

One more approval needed -- we'll need to get approval from an engineer on the Core team before this is merged.

**Review by gshank — APPROVED** [review]

**Pull request merged**
